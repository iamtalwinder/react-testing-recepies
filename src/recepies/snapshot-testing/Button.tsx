import React from 'react';

type ButtonProps = {
  text?: string;
  onClick?: () => void;
  color?: string;
  disabled?: boolean;
};

function Button({ text, onClick, color, disabled }: ButtonProps) {
  return (
    <button
      onClick={onClick}
      style={{ backgroundColor: color }}
      disabled={disabled}>
      {text}
    </button>
  );
};

export default Button;
