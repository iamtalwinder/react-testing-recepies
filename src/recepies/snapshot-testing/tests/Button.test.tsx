import { render } from '@testing-library/react';
import Button from '../Button';

describe('Button Component Snapshot Tests', () => {
  it('renders correctly with default props', () => {
    const { asFragment } = render(<Button text="Click me" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with custom color', () => {
    const { asFragment } = render(<Button text="Click me" color="blue" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const { asFragment } = render(<Button text="Click me" disabled />);
    expect(asFragment()).toMatchSnapshot();
  });
});
