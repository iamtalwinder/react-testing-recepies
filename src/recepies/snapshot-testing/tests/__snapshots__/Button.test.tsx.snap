// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshot Tests renders correctly when disabled 1`] = `
<DocumentFragment>
  <button
    disabled=""
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshot Tests renders correctly with custom color 1`] = `
<DocumentFragment>
  <button
    style="background-color: blue;"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshot Tests renders correctly with default props 1`] = `
<DocumentFragment>
  <button>
    Click me
  </button>
</DocumentFragment>
`;
